src/ch02-02-01-Starknet-devnet-rs.md

```diff
@@ -0,0 +1,370 @@
+| Private key | 0x2749bb7c72c988b035b1ef312cf3d7ef
+| Public key | 0xb7fe32817253d389251e80c1611b639e5e1cceb5761d1127d93ddd7a651304
+
+| Account address | 0x3a77277569bb8225f7aa3d7dc4e467296c9143f4f778d63ff6189d359f0c409
+| Private key | 0x87a4ed16fad5dac9988d1b2e6cd8872d
+| Public key | 0x29cb629b8d79f1efb928f553a781518ae1389c55fb61443e6087945b35fef4c
+
+| Account address | 0x49f78b29f6eee314a6c297f7a2f943a0c93dc4ec2c8f3c2089c05f351d0660b
+| Private key | 0x94de45f28d52b5808ef42bd8e5261f90
+| Public key | 0x5c7f51e81c5d1770201f6ef31e46fed058b10322478f4f1d57ae4c542a5d60e
+
+Predeployed accounts using class with hash: 0x61dac032f228abef9c6626f995015233097ae253a7f72d68552db02f2971b8f
+Initial balance of each account: 1000000000000000000000 WEI and FRI
+Seed to replicate this account sequence: 534956567
+
+## Crafting a Starknet Smart Contract
+
+**Important:** Before we proceed with this example, please ensure that the versions of both `Scarb` and `starkli` match the specified versions provided below.
+You can install the latest [scarb](#https://docs.swmansion.com/scarb/download.html) version.
+
+```console
+    rustc --version  #1.73.0 (cc66ad468)
+    Scarb --version  #  2.5.0 (c531a6e50)
+    starkli --version  # 0.1.20 (e4d2307)
+```
+
+For a more optimized and faster performance (though with a longer compilation time), run with:
+
+```
+$ cargo run --release
+```
+
+Now begin by initiating a Scarb project:
+
+```bash
+scarb new my_contract
+```
+
+## Configure Environment Variables and the `Scarb.toml` File
+
+Review the `my_contract` project. Its structure appears as:
+
+```bash
+    src/
+      lib.cairo
+    .gitignore
+    Scarb.toml
+```
+
+Amend the `Scarb.toml` file to integrate the `starknet` dependency and introduce the `starknet-contract` target:
+
+```toml
+    [dependencies]
+    starknet = ">=2.3.0"
+
+    [[target.starknet-contract]]
+     casm = true
+
+```
+
+For streamlined Starkli command execution, establish environment variables. Two primary variables are essential:
+
+- One for your account, a pre-funded account on the local development network
+- Another for designating the network, specifically the local Starknet-devnet-rs
+
+In the `src/` directory, create a `.env` file with the following:
+
+```bash
+export RPC_SPEC_VERSION="0.5.1"
+export STARKNET_VERSION="0.13.0"
+```
+
+These settings streamline Starkli command operations.
+
+## Declaring Smart Contracts in Starknet
+
+Deploying a Starknet smart contract requires two primary steps:
+
+- Declare the contract's code.
+- Deploy an instance of that declared code.
+
+Begin with the `src/lib.cairo` file, which provides a foundational template. Remove its contents and insert the following:
+
+```rust
+#[starknet::contract]
+mod hello {
+    #[storage]
+    struct Storage {
+        name: felt252,
+    }
+
+    #[constructor]
+    fn constructor(ref self: ContractState, name: felt252) {
+        self.name.write(name);
+    }
+
+    #[external(v0)]
```

Comment:
I think we have to replace #[external(v0)] with something new

Reply:
can we replace with this #[abi(embed_v0)] ?

Reply:
yes